corehq/apps/hqadmin/views/users.py

```diff
@@ -97,12 +97,22 @@ def post(self, request, *args, **kwargs):
             for user in users:
                 # save user object only if needed and just once
                 if user.is_superuser is not is_superuser:
+                    update_log = {
+                        'field': 'is_superuser',
+                        'previous': user.is_superuser,
+                        'current': is_superuser,
+                    }
                     user.is_superuser = is_superuser
-                    changed_fields.append('is_superuser')
+                    changed_fields.append(update_log)
 
                 if can_toggle_is_staff and user.is_staff is not is_staff:
+                    update_log = {
+                        'field': 'is_staff',
+                        'previous': user.is_staff,
```

Comment:
Can we avoid the previous value? It seems overkill since the value would be present in the last log?

More context,
On USH, we are working on tracking user updates, where, as of now the plan is to just keep note of the fields changed, with their current value. So it would more in line with that process.

Is there a scenario, where the previous value will be helpful?

Reply:
Addressed in https://github.com/dimagi/commcare-hq/pull/29787#issuecomment-849567284

Reply:
thank you. I would request for just the current value / current state (as a text). More in the comment on the PR
Will let @dannyroberts share if that is okay from what the perspective of what he had in mind when he made the ticket.

Reply:
The only thing if say is that someone's redundancy in a log can help you tell if something's wrong/missing. Let's say there were some other way to update the superuser field that for some reason does not result in a log entry. Since we expect the previous value to always match the previous event's current value, we would be able to tell if there was a historical gap in the logs. In the other hand seeing the history match up gives some confidence that you're seeing the full story. Anyway, not sure how crucial that is, but I just wanted to make a case for a little redundancy in logs like this

Reply:
Oh sorry I just understood that we're talking only about Boolean values here. As long as we only mention fields that changed, since there are only two values (true/false), knowing that the current value and knowing that it changed means you also know the previous value. If all that is right then what I said above doesn't really apply here.

Reply:
Thanks @dannyroberts.

`if user.is_superuser is not is_superuser`
does mean that we are updating only when there is a change, so does make the previous value redundant.

It's true that if there are other places where we are not logging updates, there would be a gap in the history. So we just need to try and cover up all places where an update happens, which we are hoping to achieve in our work on USH.